lib/transformer.js

```diff
@@ -0,0 +1,111 @@
+'use strict'
+
+const zlib = require('zlib')
+const aws = require('aws-sdk')
+const {
+  checkFileFormat
+, getProperty
+, hasProperty
+} = require('./utils.js')
+
+const s3 = new aws.S3({
+  apiVersion: '2006-03-01'
+})
+
+module.exports = {
+  extractData
+, getLogs
+, prepareLogs
+}
+
+function extractData(data) {
+  if (!data) return []
+
+  const {
+    Records
+  , logFiles
+  , ...meta
+  } = data
+
+  let ts = Date.now()
+  if (hasProperty(meta, 'eventTime')) {
+    ts = (new Date(getProperty(meta, 'eventTime'))).getTime()
+  }
+
+  let records = data
+  if (Records) {
+    records = Records
+  } else if (logFiles) {
+    records = logFiles
+  }
+
+  if (!Array.isArray(records)) {
+    return [{
+      line: JSON.stringify(records)
+    , meta
+    , timestamp: ts
+    }]
+  }
+
+  return records.map((record) => {
+    const lineJSON = {
+      line: JSON.stringify(record)
+    , meta
+    , timestamp: ts
+    }
+
+    if (hasProperty(record, 'eventTime')) {
+      lineJSON.timestamp = (new Date(getProperty(record, 'eventTime'))).getTime()
+    }
+
+    return lineJSON
+  })
+}
+
+function getLogs(params, callback) {
+  const keyFormat = checkFileFormat(params.Key)
+  return s3.getObject(params, (error, data) => {
+    if (error) {
+      return callback(`Error in Getting ${params.Key} from ${params.Bucket}: ${error}`)
+    }
+
+    let content = data.Body
+    if (keyFormat.gz) {
+      content = zlib.gunzipSync(content.Body)
+    }
+
+    content = content.toString('ascii')
+    if (keyFormat.json) {
+      try {
+        content = JSON.parse(content)
+        return callback(null, extractData(content))
+      } catch (e) {
+        return callback(`Error in Parsing the Data from JSON file ${params.Key}: ${e}`)
+      }
+    }
+
+    return callback(null, content.split('\n').map((line) => {
+      return {line}
+    }))
+  })
+}
+
+function prepareLogs(logs, eventData) {
+  return logs.filter((log) => {
+    return log !== '' || !log.line
```

Comment:
is `log` expected to be either a string _or_ an object? it doesn't look like that is the case below. maybe just:

```js
return !log || !log.line
```

Reply:
It's expected to be an object